sml/src/sml_transport.c

```diff
@@ -33,36 +33,18 @@ unsigned char start_seq[] = {0x1b, 0x1b, 0x1b, 0x1b, 0x01, 0x01, 0x01, 0x01};
 unsigned char end_seq[] = {0x1b, 0x1b, 0x1b, 0x1b, 0x1a};
 
 
-size_t sml_read(int fd, fd_set *set, unsigned char *buffer, size_t len) {
-	
-	ssize_t r = 0;
-	size_t tr = 0;
-
-	while (tr < len) {
-		select(fd + 1, set, 0, 0, 0);
-		if (FD_ISSET(fd, set)) {
-			
-			r = read(fd, &(buffer[tr]), len - tr);
-			if (r < 0) continue;
-
-			tr += r;
-		}
-	}
-	return tr;
-}
-
-size_t sml_transport_read(int fd, unsigned char *buffer, size_t max_len) {
-
-	fd_set readfds;
-	FD_ZERO(&readfds);
-	FD_SET(fd, &readfds);
+int sml_transport_read(int fd, unsigned char *buffer, size_t max_len) {
 
+	int bytes;
 	unsigned char buf[max_len];
```

Comment:
this might need a big stack and is quite risky if somebody calls with big max_len. Why not write directly into the provided buffer?

Reply:
I really dunno- existing code...